### PR TITLE
Carbon Calculator model change and method testing

### PR DESCRIPTION
**Pull request by BradHN1**
This PR addresses two issues:
API #102 - Carbon Calculator - Longer description text for actions and stations
API #133 - Carbon Calculator - Range for numeric parameters
It also adds the carbon calculator method testing

### CODE REVIEW

File: src/_main_/settings.py [hunk 1]
```diff
@@ -92,19 +92,23 @@
...
+AWS_STORAGE_BUCKET_NAME  = os.environ.get('AWS_STORAGE_BUCKET_NAME')
 AWS_S3_SIGNATURE_VERSION = os.environ.get('AWS_S3_SIGNATURE_VERSION')
-AWS_S3_REGION_NAME = os.environ.get('AWS_S3_REGION_NAME')
-AWS_DEFAULT_ACL  = None
+AWS_S3_REGION_NAME       = os.environ.get('AWS_S3_REGION_NAME')
+AWS_DEFAULT_ACL          = None
 #--------END AWS CONFIGURATION ---------------------#
 
+
+# @Sam - does this cause any problems?
```
Comment (massenergize-admin) [inline src/_main_/settings.py:109]: i think its fine.  Remember to remove the comment though lol

File: src/_main_/settings.py [hunk 2]
```diff
@@ -137,21 +141,21 @@
...
 # https://docs.djangoproject.com/en/2.1/ref/settings/#databases
 DATABASES = {
-    'default': {
-        'ENGINE': os.environ.get('DATABASE_ENGINE'),
-        'NAME': os.environ.get('DATABASE_NAME'),
-        'USER': os.environ.get('DATABASE_USER'),
-        'PASSWORD': os.environ.get('DATABASE_PASSWORD'),
-        'HOST': os.environ.get('DATABASE_HOST'),
-        'PORT': os.environ.get('DATABASE_PORT')
+    'remote-default': {
```
Comment (massenergize-admin) [inline src/_main_/settings.py:144]: again, remove.  since you have the local.env we should be fine

File: src/carbon_calculator/CCConstants.py [hunk 3]
```diff
@@ -9,3 +9,9 @@
 HEATING_SYSTEM_POINTS = 10000
 SOLAR_POINTS = 6000
 ELECTRICITY_POINTS = 5000
+
+BTU_PER_KWH = 3414              # https://www.eia.gov/energyexplained/units-and-calculators/
+BTU_PER_THERM = 100000.
+BTU_PER_GAL_FUELOIL = 137619    # https://www.eia.gov/energyexplained/units-and-calculators/
+BTU_PER_GAL_GASOLINE = 120286   # https://www.eia.gov/energyexplained/units-and-calculators/
+BTU_PER_GAL_PROPANE = 91333     # https://www.eia.gov/energyexplained/units-and-calculators/
```
Comment (massenergize-admin) [inline src/carbon_calculator/CCConstants.py:17]: add new line

File: src/carbon_calculator/CCDefaults.py [hunk 4]
```diff
@@ -93,28 +123,71 @@ def importDefaults(self,fileName):
...
             status = False
-
-        if csvfile:
-            csvfile.close()
-        return status
\ No newline at end of file
+        finally:
+            if csvfile:
+                csvfile.close()
+            return status
```
Comment (massenergize-admin) [inline src/carbon_calculator/CCDefaults.py:193]: add new line

File: src/carbon_calculator/content/Events.csv [hunk 5]
```diff
@@ -1,4 +1,8 @@
...
-,,,,,,,,,,,,,,
\ No newline at end of file
+﻿Name,DisplayName,DateTime,Location,Stations (in reverse order in AirTable),Groups,Host organization,Sponsor organization,VisibleInCarbonSaver,Question_group
+CC_Event_1,Cooler Communities event,4/22/2020 1:00pm,High School somewhere,"Welcome_1,Heating_Cooling_1,Water_Heating_1,Solar_1,Electricity_Use_1,Transportation_1,Eating_1,Reduce_Reuse_1,Yard_Landscaping_1","the_high_school,the_middle_school,one_elementary_school,another_elementary_school",Ener-G-Save,Harold Grinspoon Foundation,checked,
+Hadley,Cooler Communities Hadley,5/21/2020 5:00pm,"Hopkins Academy, 131 Russell St, Hadley, MA 01035","Heating_Cooling_CCH,Welcome_CCH,Electricity_CCH,Solar_CCH,Transportation_CCH,Eating_CCH,Reduce_Reuse_CCH",,Hadley Public Schools,Cooler Communities,checked,Hadley
+MEPlatform,MassEnergize Community Platform,,,All_Actions_ME,,MassEnergize,,,ME
+,,,,,,,,,
+,,,,,,,,,
+,,,,,,,,,
+,,,,,,,,,
```
Comment (massenergize-admin) [inline src/carbon_calculator/content/Events.csv:8]: add new line

File: src/carbon_calculator/content/Stations.csv [hunk 6]
```diff
@@ -1,11 +1,20 @@
...
+Eating_CCH,Eating,"A low carbon diet is also a healthy diet, both for you and the planet",eating.png (https://dl.airtable.com/.attachments/f997b1f6db475b5c7cb6d791db19307c/3c46845e/eating.png),Hadley,low_carbon_diet,
+Reduce_Reuse_CCH,Reduce/Reuse,"Recycling is good, but reducing packaging and consumption has even more impact.",recycle.png (https://dl.airtable.com/.attachments/c60d3d4b723185db511df9dd0391f74e/a805fa2b/recycle.png),Hadley,"reduce_waste,compost",
+All_Actions_ME,All Actions,This is a group which includes all supported actions for the MassEnergize Community Platform,,MEPlatform,"energy_fair,energy_audit,prog_thermostats,weatherization,community_solar,renewable_elec,led_lighting,heating_assessment,efficient_fossil,air_source_hp,ground_source_hp,hw_assessment,hp_water_heater,solar_assessment,install_solarPV,install_solarHW,energystar_fridge,energystar_washer,induction_stove,hp_dryer,coldwater_wash,line_dry,fridge_pickup,smart_power_strip,electricity_monitor,replace_car,reduce_car_miles,eliminate_car,reduce_flights,offset_flights,low_carbon_diet,reduce_waste,compost,reduce_lawn_size,reduce_lawn_care,electric_mower,rake_elec_blower",
+,,,,,,
 ,,,,,,
```
Comment (massenergize-admin) [inline src/carbon_calculator/content/Stations.csv:20]: same here, new line

File: src/_main_/settings.py (hunk 1 above)
Comment (BradHN1) [reply src/_main_/settings.py:109]: ok

File: src/_main_/settings.py (hunk 2 above)
Comment (BradHN1) [reply src/_main_/settings.py:144]: ok

File: src/carbon_calculator/CCConstants.py (hunk 3 above)
Comment (BradHN1) [reply src/carbon_calculator/CCConstants.py:17]: ok

File: src/carbon_calculator/CCDefaults.py (hunk 4 above)
Comment (BradHN1) [reply src/carbon_calculator/CCDefaults.py:193]: ok

File: src/carbon_calculator/content/Events.csv (hunk 5 above)
Comment (BradHN1) [reply src/carbon_calculator/content/Events.csv:8]: ok

File: src/carbon_calculator/content/Stations.csv (hunk 6 above)
Comment (BradHN1) [reply src/carbon_calculator/content/Stations.csv:20]: These are exported from another program

**Pull request merged**